### PR TITLE
RenameRefactoring re-design

### PR DESCRIPTION
**Pull request by BZngr**
This PR is a re-designed version of RenameRefactoring.cs.  The new design replaces my prior PR's interface-based approach with a delegate approach since the interface essentially defined a single-method.  This resulted in deletion of all the interface implementation class files.  And, hopefully, a more readable/clearer implementation.  The RenameRefactoring tests were also refactored to reduce boilerplate code.  The redesign also exposed some bugs that were fixed along the way.

RenameRefactoring bugs/enhancements:
1. WithEvents field renames did not rename member Eventhandler names
2. EventHandlers used within other members were not renamed 
3. Removed all useages of 'AllUserDeclarations'
4. Renaming a ClassModule that defines an interface by selecting the class name at an "Implements" statement did not successfully rename all other references to the interface/class.
5. User is now prevented from attempting to rename UserForm_*, Class_*, auto_* event handlers. 
6. The User pop-up continue/confirmation strategy for interfaces is now applied in a consistent manner for event and control renames.

Note:
Control Rename Tests: All the tests for Control renames are currently marked as "Ignore".  This is because the logic depends upon calling the DeclarationFinder.FindAllEventHandlers() to retrieve built-in events.  I'm wondering if these tests should be removed - or is there a possibility of one-day mocking the DeclarationFinder to support built-in types?

### CODE REVIEW

**Comment by Hosch250** [comment]
You can create the built-in declarations through the existing ctor, but it is a bit of a pain sometimes.

File: RetailCoder.VBE/Refactorings/Rename/RenameRefactoring.cs [hunk 1]
```diff
@@ -10,357 +9,558 @@
...
 
-        private bool IsControlEventHandler(Declaration userTarget, out Declaration control)
+        private void RenameProject()
         {
-            control = null;
-            var declarationsOfInterest = _model.State.AllUserDeclarations.Where(d => d.DeclarationType.HasFlag(DeclarationType.Control)
-                    && userTarget.IdentifierName.StartsWith(d.IdentifierName));
+            RequestParseAfterRename = false;
+            var projects = _model.VBE.VBProjects;
+            var project = projects.SingleOrDefault(p => p.HelpFile == _model.Target.ProjectId);
```
Comment (retailcoder) [inline RetailCoder.VBE/Refactorings/Rename/RenameRefactoring.cs]: `HelpFile` is an implementation detail of `ProjectId`; wouldn't it be better to compare `p.ProjectId` instead? That way if we change the way we ID a project then that bit won't break...

Comment (BZngr) [inline RetailCoder.VBE/Refactorings/Rename/RenameRefactoring.cs]: Updated as indicated.  Yeah....I thought the use of `HelpFile` in the logic looked a little odd.  But, I pretty much took this function as I found it and incorporated it within the structural changes.  That said, using `p.ProjectId` works just as well, looks more intuitive - and, from your comment, sounds less likely to break.  TBH, I am not sure how the user can select and initiate a project rename.  The unit test code for renaming a project does not initiate using a selection like the other tests.   And, within Excel, I could not figure out a way to cause this code to execute.

Comment (retailcoder) [inline RetailCoder.VBE/Refactorings/Rename/RenameRefactoring.cs]: The command is (or, was and should still be) exposed in the Code and Project Explorer toolwindows' respective context menus, when you right-click a project node ;-)

**Pull request merged**